### PR TITLE
Example that builds a Flutter Android app using Gradle

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

File: examples/gradle/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+This project demonstrates how to embed Flutter within an Android application
+and build the Android and Flutter components with Gradle.
+
+To build this project:
+
+* Create a `local.properties` file with these entries:
+  * `sdk.dir=[path to the Android SDK]`
+  * `flutter.sdk=[path to the Flutter SDK]`
+  * `flutter.engine=[path to the flutter.jar file in your build of the Flutter engine]`
```
Comment (abarth) [inline examples/gradle/README.md]: Maybe call this flutter.jar ?

**Comment by abarth** [comment]
This looks great.  We might want to call the example something other than "gradle" because that's a string that's required in other parts of the example.  If we pick a more fanciful name, it will be clearer which of the strings "gradle" a customer can replace with their own string.

**Comment by abarth** [comment]
LGTM w/ a different directory name.  Maybe hello_android?

File: examples/gradle/README.md (hunk 1 above)
Comment (jason-simmons) [inline examples/gradle/README.md]: done

**Comment by jason-simmons** [comment]
Renamed to hello_android

**Comment by abarth** [comment]
LGTM

**Comment by tain335** [comment]
How to gen fluter.jar file?

**Comment by abarth** [comment]
It looks like currently you need to build your own engine to create flutter.jar:

https://github.com/flutter/engine/blob/master/CONTRIBUTING.md

When you build for Android, the jar is produced in `out/android_Debug/flutter.jar`.  I've filed https://github.com/flutter/flutter/issues/3233 about including a pre-built flutter.jar in the engine artifacts.

**Comment by tain335** [comment]
I  have successed to gen flutter.jar and run Hello_Android perfectly, it is so exciting ,but I found the apk is too big(19M), I hope it could be smaller.

**Comment by abarth** [comment]
19M sounds rather large.  I would have expected something closer to 8M.  Would you be willing to file an issue about the size?  We might be ending up with multiple copies of some resources somehow:

https://github.com/flutter/flutter/issues/new

**Pull request merged**